be/src/fs/fs_posix.cpp

```diff
@@ -261,7 +261,9 @@ class PosixWritableFile : public WritableFile {
     Status sync() override {
         if (_pending_sync) {
             _pending_sync = false;
-            RETURN_IF_ERROR(do_sync(_fd, _filename));
+            if (config::sync_tablet_meta) {
+                RETURN_IF_ERROR(do_sync(_fd, _filename));
+            }
```

Comment:
this config should not affect the behavior of posix_fs(it affects all file writes), it should be checked when close segment file I think